### ISSUE TITLE
On the Next Episode - Black screen after clicking play on preloader

### ISSUE THREAD

**Issue by TomFulp** [ruffle-rs/ruffle#2287]
https://www.newgrounds.com/portal/view/600444/format/flash?emulate=flash

The preloader screen for this entry appears but when you click play, it goes to a black screen.

**Comment by Poopooracoocoo** [comment]
I also saw a black screen with The Binding of Isaac: Wrath of the Lamb.

**Comment by winston895** [comment]
I have tested the game with the current Ruffle Jan. 6, 2021 plugin for FireFox. I clicked on the play button in the middle of the screen with the boy holding up his left fist, the screen goes to a black page for about 1 second, then it goes to a white page 1 second later, and then goes to the main screen with the black and grey colored tank with the [play] button at the bottom right corner. After clicking on the [play] button, the screen turns.  
      BUT
This is the exact thing that happens when I used Waterfox with FlashPlayer version 28. So, is it possible that the game is currupt? I believe that it is NOT a problem with Ruffle emulation.

**Comment by winston895** [comment]
![Ruffle with Firefox plugin Jan 6 2021](https://user-images.githubusercontent.com/70994584/103815079-d2c96b80-5030-11eb-85f2-f14db8260526.jpg)
This is the picture of the black/gray tank, using Ruffle with Firefox plugin.

**Comment by Toad06** [comment]
The issue is located in `\DefineSprite (19: Preloader)\frame 10\DoAction`:

```
function _onPlayClick()
{
   _root.nextFrame();
   // Execution stops here in Ruffle, continues till the end of function in Flash.
   _root.play();
   _parent._onLoaded();
}
```

**Comment by Herschel** [comment]
Looks like the "stop AS immediately if clip is removed" should not apply to buttons in this case.

**Comment by Herschel** [comment]
Looked into it more, and it's a complex case of #1535.

The Flash Ad component has an `unload` event handler. Having an unload handler causes a clip (and all ancestor clips) to survive for one frame longer, so that the unload can properly run. This allows the play button to survive and run the rest of the code in the function. (This must have totally eluded me when I wrote the NG preloader code!)

If you delete the Flash Ad from the preloader, or use the play button by itself, the buggy behavior occurs in both Flash and Ruffle.

Here's a simplified example. A clip contains a child clip with an unload, and a button that plays. If you remove the unload, the play doesn't occur.
[button-unload.zip](https://github.com/ruffle-rs/ruffle/files/5789444/button-unload.zip)

**Comment by waspennator** [comment]
Tested this out on the latest nightly build with #8873 merged in and it seems to be capable of getting past the black screen now on desktop and no longer needs the right click and play trick on web using the ruffle web demo.

However, the animation still has one more visual issue, for some reason goku, vegeta, etc have these lines portruding out from them.

On Ruffle Nightly 2/7/23

![On the next episode of Ruffle](https://user-images.githubusercontent.com/78768499/217121163-dd504956-ce40-4371-a214-42c1027b58a5.png)

On Flash

![On the next episode of Flash](https://user-images.githubusercontent.com/78768499/217121456-b83e33c2-202e-4ca5-85a2-9902b0a9f450.png)

**Comment by Lord-McSweeney** [comment]
Yeah, that's the lyon bug. Seeing as that should be fixed with the latest lyon update, I'm closing this issue.

**Comment by waspennator** [comment]
Visual line issues have also been fixed now with #9499 merged in.
